python/graphstorm/dataloading/dataset.py

```diff
@@ -156,6 +186,72 @@ def edge_feat_field(self):
         """the field of edge feature"""
         return self._edge_feat_field
 
+    def load_wg_feat(self, part_config, num_parts, ntype, rank, num_ranks):
+        """Load features from wholegraph memory
+
+        Parameters
+        ----------
+        part_config : str
+            The path of the partition configuration file.
+        num_parts : int
+            The number of partitions of the dataset
+        ntype: str
+            The type of node for which to fetch features.
+        rank: int
+            The rank of the process
+        num_ranks: int
+            The total number of ranks (world size)
+        """
+        import pylibwholegraph.torch as wgth
+        import pylibwholegraph.binding.wholememory_binding as wmb
+
+        if not self._wg_init:
+            class Options:
+                """Set Wholegraph's environment variables """
+            Options.launch_agent = 'pytorch'
+            Options.launch_env_name_world_rank = 'RANK'
+            Options.launch_env_name_world_size = 'WORLD_SIZE'
+            Options.launch_env_name_local_rank = 'LOCAL_RANK'
+            Options.launch_env_name_local_size = 'LOCAL_WORLD_SIZE'
+            Options.launch_env_name_master_addr = 'MASTER_ADDR'
+            Options.launch_env_name_master_port = 'MASTER_PORT'
+            Options.local_rank = rank % role.get_num_trainers()
+            Options.local_size = role.get_num_trainers()
+
+            wgth.distributed_launch(Options, lambda: None)
+            wmb.init(0)
+            th.set_num_threads(1)
+            wgth.comm.set_world_info(rank, num_ranks, Options.local_rank, Options.local_size)
+            self._wg_init = True
+        global_comm = wgth.comm.get_global_communicator()
+
+        feature_comm = global_comm
+        embedding_wholememory_type = 'distributed'
+        embedding_wholememory_location = 'cpu'
+        cache_policy = wgth.create_builtin_cache_policy(
+            "none", # cache type
+            embedding_wholememory_type,
+            embedding_wholememory_location,
+            "readonly", # access type
+            0.0, # cache ratio
+        )
+        metadata_file = os.path.join(os.path.dirname(part_config), 'wholegraph/metadata.json')
+        with open(metadata_file) as f:
+            wg_metadata = json.load(f)
+        node_feat_wm_embedding = wgth.create_embedding(
+            feature_comm,
+            embedding_wholememory_type,
+            embedding_wholememory_location,
+            getattr(th, wg_metadata[ntype + '/feat']['dtype'].split('.')[1]),
+            wg_metadata[ntype + '/feat']['shape'],
+            optimizer=None,
+            cache_policy=cache_policy,
+        )
+        feat_path = os.path.join(os.path.dirname(part_config), 'wholegraph', ntype + '~feat')
```

Comment:
Nit: I guess Nicolas's script for dumping feature metafile/binary files, should be included in the repo as a separate PR for completeness.

Reply:
Here's the PR: https://github.com/awslabs/graphstorm/pull/428